src/util/upnp_clients.cc

```diff
@@ -255,18 +203,75 @@ bool Clients::getInfoByAddr(const struct sockaddr_storage* addr, const ClientInf
 
 bool Clients::getInfoByType(const std::string& match, ClientMatchType type, const ClientInfo** ppInfo)
 {
-    auto it = std::find_if(clientInfo.begin(), clientInfo.end(), [=](const auto& c) //
-        { return c.matchType == type && match.find(c.match) != std::string::npos; });
+    if (!match.empty()) {
+        auto it = std::find_if(clientInfo.begin(), clientInfo.end(), [=](const auto& c) //
+            { return c.matchType == type && match.find(c.match) != std::string::npos; });
+
+        if (it != clientInfo.end()) {
+            *ppInfo = &(*it);
+#if TOMBDEBUG
+            log_debug("found client by type (match='{}')", match.c_str());
```

Comment:
log_debug get compiled out if debug is off anyway, so dont need the #if

Reply:
my idea first hand was to not have log_debug, as is not log_trace i made #if 0 around it. what shall we do now?

Reply:
ok the more debug info the better, i just removed the #if